hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusSubmitMessageHandler.java

```diff
@@ -43,20 +43,30 @@ public ConsensusSubmitMessageHandler() {}
     /**
      * This method is called during the pre-handle workflow.
      *
-     * <p>Typically, this method validates the {@link TransactionBody} semantically, gathers all
-     * required keys, warms the cache, and creates the {@link TransactionMetadata} that is used in
-     * the handle stage.
-     *
-     * <p>Please note: the method signature is just a placeholder which is most likely going to
-     * change.
+     * <p>Determines signatures needed for submitting a new message to a consensus topic
      *
      * @param context the {@link PreHandleContext} which collects all information that will be
      *     passed to {@code handle()}
+     * @param topicStore the {@link ReadableTopicStore} to use to resolve topic metadata
      * @throws NullPointerException if one of the arguments is {@code null}
      */
-    public void preHandle(@NonNull final PreHandleContext context) {
+    public void preHandle(@NonNull final PreHandleContext context, @NonNull ReadableTopicStore topicStore) {
         requireNonNull(context);
-        throw new UnsupportedOperationException("Not implemented");
+        requireNonNull(topicStore);
+
+        final var op = context.getTxn().getConsensusSubmitMessage();
+        final var topicMeta = topicStore.getTopicMetadata(op.getTopicID());
+        if (topicMeta.failed()) {
+            context.status(ResponseCodeEnum.INVALID_TOPIC_ID);
+            return;
+        }
+
+        final var submitKey = topicMeta.metadata().submitKey();
+        if (submitKey.isEmpty()) {
```

Comment:
This isn't current behavior, is it?

Reply:
Yep you're right, not sure where I got this from. Good catch, thanks!